src/PowerShell/PowerShellManagerPool.cs

```diff
@@ -46,6 +46,18 @@ internal PowerShellManagerPool(MessagingStream msgStream)
             RpcLogger.WriteSystemLog(string.Format(PowerShellWorkerStrings.LogConcurrencyUpperBound, _upperBound.ToString()));
         }
 
+        /// <summary>
+        /// Initialize the pool and populate it with PowerShellManager instances.
+        /// We instantiate PowerShellManager instances in a lazy way, starting from size 1 and increase the number of workers as needed.
+        /// </summary>
+        internal void Initialize(PowerShell pwsh)
+        {
+            var logger = new RpcLogger(_msgStream);
+            var psManager = new PowerShellManager(logger, pwsh);
+            _pool.Add(psManager);
+            _poolSize = 1;
```

Comment:
I would consider doing this as _poolSize++

Reply:
This is the initialize method. It only gets called once so setting it to 1 makes sense. 

He's doing the thread safe increment, `Interlocked.Increment(ref _poolSize)` (since `++` isn't thread safe), on line 76 below

Reply:
The assignment can also be overwritten by another thread, but since internal void Initialize(PowerShell pwsh) is called from the FunctionLoad, so Interlocked.Increment(ref _poolSize) can be used here

Reply:
`PowerShellManagerPool.Initialize` is supposed to add the very first PowerShellManager instance to the pool, and it shouldn't ever be invoked by multiple threads.
The following comment message needs to be updated though. I will push a commit to update it.
> /// Initialize the pool and populate it with PowerShellManager instances.